flask_security/forms.py

```diff
@@ -125,7 +125,8 @@ class NewPasswordFormMixin():
 class PasswordConfirmFormMixin():
     password_confirm = PasswordField(
         get_form_field_label('retype_password'),
-        validators=[EqualTo('password', message='RETYPE_PASSWORD_MISMATCH')])
+        validators=[password_required,
```

Comment:
IMHO `password_required` is not needed since the password must be equal to `password` field which is already required.